src/cephadm/cephadm

```diff
@@ -627,26 +627,27 @@ def infer_fsid(func):
     @wraps(func)
     def _infer_fsid():
         if args.fsid:
-            logger.debug('Using specified fsid: %s' % args.fsid)
+            logger.info('Using specified fsid: %s' % args.fsid)
             return func()
 
-        fsid_list = []
-        if os.path.exists(args.data_dir):
-            for i in os.listdir(args.data_dir):
-                if is_fsid(i):
-                    fsid_list.append(i)
-        logger.debug('Found fsids %s' % str(fsid_list))
-
-        if not fsid_list:
-            # TODO: raise?
-            return func()
-
-        if len(fsid_list) > 1:
+        fsids = set()
+        daemon_list = list_daemons(detail=False)
+        for daemon in daemon_list:
+            if 'name' not in args or not args.name:
+                fsids.add(daemon['fsid'])
+            elif daemon['name'] == args.name:
+                fsids.add(daemon['fsid'])
+        fsids = list(fsids)
+        logger.info('Found fsids %s' % str(fsids))
+
+        if len(fsids) == 1:
+            logger.info('Inferring fsid %s' % fsids[0])
+            args.fsid = fsids[0]
+        elif len(fsids) > 1:
             raise Error('cannot infer fsid, must specify --fsid')
```

Comment:
and add it here, something like ``f"ERROR: unable to infer fsid, must specify --fsid ({name} exists in fsids {fsids})"``

Reply:
The `name` arg is optional. Maybe this instead?

```
ERROR: Cannot infer an fsid, one must be specified: ['2e8c319a-3d73-11ea-b7d6-8cdcd44b8f38', '3eb4399a-3d6f-11ea-ba23-8cdcd44b8f38']
```